README.md

```diff
@@ -1 +1,679 @@
+
+```fsharp
+success_vars
+    |> readLines
+    |> Seq.map tokenize
+    |> Seq.choose each
+    |> Seq.filter (keyToUpper >> keyContains "ELEMENT")
+    |> Seq.sort 
+    |> Seq.map (values >> toInt)
+    |> Seq.fold sum (Some 0)
+    >>= format
+    >>= printValue
+```
+
+As you might remember, the `>>=` is a bind operation. Bind can be seen as a wrapper function that calls the function if the argument has a value or doesn't if the value is none.
+
+In code, it looks like this:
+
+```fsharp 
+let bind f optional =
+    match optional with
+    | Some x -> f x
+    | _ -> None
+```
+
+There are additional implementations in f#, for instance, the `choose` function on sequences we have already used. Thus, this behaviour can be assumed as law in every functional realm. Knowing this law and combining it with other elements like lazy evaluations and constructs like `map` or `either` will enable you to write complex computational flows that can be mixed and sequenced freely and predictable.
+
+For instance, if we are interested in the fact that we ended up with `None` in our composition, we can change the code like so:
+
+```fsharp
+let printValue (value: string option) =
+    match value with
+        | Some v -> v |> printfn "%s"
+        | _ -> printfn "No value available"
+
+success_vars
+    |> readLines
+    |> Seq.map tokenize
+    |> Seq.choose each
+    |> Seq.filter (keyToUpper >> keyContains "ELEMENT")
+    |> Seq.sort 
+    |> Seq.map (values >> toInt)
+    |> Seq.fold sum (Some 0)
+    >>= format
+    |> printValue
+```
+
+The result is now evaluated as:
+
+```sh
+❯ dotnet fsi pipes.fsx
+No value available
+❯ 
+```
+
+More importantly, though, when wrapped in a bind, our function will only be called given a valid, existing value. We can trust our computational flow to handle the data; thus, we can focus on the logic.
+
+And this is where things start to fall apart in javascript.
+
+Remember our previous example in javascript, and change the code as we did in f#.
+
+```js
+let test_data = `
+ELEMENT01=3
+ELEMENT04=4
+discarded:incorrect
+ELEMENT03=1
+foo=4
+element06=not an integer
+ELEMENT07=12
+bar=3
+ELEMENT02=12
+`
+test_data
+    |> readLines(^)
+    |> tokenize(^)
+    |> ^.filter(e => e |> keyToUpper(^) |> keyContains("ELEMENT", ^))
+    |> ^.sort(sort)
+    |> ^.reduce(sumValues, 0)
+    |> format(^)
+    |> console.log(^)
+```
+
+When we execute this, the result will be:
+
+```sh
+Done
+```
+
+This output is because our computational flow is broken starting from the `reduce` statement. In the `sumValues` function, `element06` will become `NaN`, and all sums after that will be `NaN`. In our format function, as we didn't account for `NaN`, we will return the result of the `else` statement, and the `console.log` is without an option to decide if the result is valid and thus should be printed.
```

Comment:
There exist the term [Railway Oriented Programming](https://fsharpforfunandprofit.com/posts/recipe-part2/)  by the brilliant Scott Wlaschin. I add a link to it, otherwise I might not be able to hold myself from adding a blogpost inside the blogpost :D